hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/reader/entities/DefaultTraceEntityReader.java

```diff
@@ -66,18 +64,18 @@
                     .map(Collections::unmodifiableMap));
   }
 
-  private Maybe<Entity> getOrCreateAvroEntity(
-      EntityType entityType, StructuredTrace trace, Event span) {
+  private Maybe<Entity> getOrCreateAvroEntity(EntityType entityType, T trace, S span) {
```

Comment:
given that these are still working within the data model (`convertToAvroEntity`), does it make sense to parameterize this with an `E` too? That would mean taking in an entity converter. Or maybe better, change the interface to work in entity service entities and keep an avro entity converter as a separate util.